### ISSUE TITLE
Reset carousel back to the first slide (index[0])

### ISSUE THREAD

**Issue by DanielNoonan** [express-labs/pure-react-carousel#116]
I have a menu and am changing the contents of the carousel depending on which menu item is clicked.

The issue I have is that when a new menu item is clicked it loads the new content but remains displaying the new content slide at the last selected index value.

Example: 
i) User chooses menu item 1 and the carousel loads the appropriate slides.
ii) User navigates through the slides to say the 4th slide (index 3).
iii) User chooses a new menu item e.g. menu item 2.
iv) The new carousel items load but the one being displayed in the 4th slide (index 3)

So I need to reset the carousel back to position index[0] whenever a new menu item is selected.

Is there a way of doing this currently. If not, it would be great if it could be added.

Thanks!

**Comment by mrbinky3000** [comment]
Yep.  This is something we ran into here at Express as well.  It's not necessarily a "bug" per se.  You don't want the slides reseting to the first slide every time the carousel re-renders.  However, it is confusing.

In our case, we wanted to set the slides back to zero whenever the user selected a new color or a new model.  To do that, we made a little Component that uses the `WithProvider()` HOC to "listen" for changes, and then use the setStoreState() method to set the `currentSlide` property back to 0.

I understand that explanation might be a little opaque if you're not familiar with the carousel.  I provide it to you just in case you do understand what I mean. 

TL;DR.  We should make a new component called "AutoReset" or "Listener" where you feed it two things to compare and if they change, it set the slide to slide 0 (or another slide specified)  

I might be able to bash that out today or tomorrow.

If anyone has a better idea for the component name, I'm listening :-)

**Comment by DanielNoonan** [comment]
Thanks for the reply, 

I'm not too familiar with this carousel as just started using it today. What I did to get around my issue (which is probably a bit 'hacky') is I gave the carouselProvider component a key prop. I then change this key's value when I want to trigger the carousel resetting back to it's default starting condition. As React uses unique keys to identify components it forces the recreation of the carousel which in turn resets the starting position.

To have a setting/prop built in that can be updated to move the carousel to a particular index value would be a very welcome addition.

**Comment by Alek997** [comment]
maybe this can help you
https://github.com/express-labs/pure-react-carousel/issues/60#issuecomment-485275619
